lib/_stream_readable.js

```diff
@@ -1070,13 +1072,84 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
-  if (createReadableStreamAsyncIterator === undefined) {
-    createReadableStreamAsyncIterator =
-      require('internal/streams/async_iterator');
+  let stream = this;
+
+  if (typeof stream.read !== 'function') {
+    // v1 stream
+    const src = stream;
+    stream = new Readable({
+      objectMode: true,
+      destroy(err, callback) {
+        destroyImpl.destroyer(src, err);
+        callback();
+      }
+    }).wrap(src);
   }
-  return createReadableStreamAsyncIterator(this);
+
+  const iter = createAsyncIterator(stream);
+  iter.stream = stream;
+  return iter;
 };
 
+async function* createAsyncIterator(stream) {
+  const state = stream._readableState;
+
+  if ((state && state.endEmitted) ||
+      (state && state.closeEmitted)) {
+    // TODO (ronag): What if errorEmitted?
+    // TODO (ronag): Wait for closeEmitted?
+    // TODO (ronag): if (closeEmitted && !endEmitted) ERR_PREMATURE_CLOSE?
+    stream.on('error', nop);
+    return;
+  }
+
+  let callback = nop;
```

Comment:
I think this implementation is mostly correct in terms of error/readable timings.

Just timings when reviewing of readable/error:
```
 - `readable` gets dispatched
 - callback called and set to nop
 - promise fulfills, callback gets set
 - `error` gets emitted, callback gets called with the right value for error ✔️ 

 - `readable` gets dispatched, callback gets called and set to noop
 - `error` gets emitted, the error gets set and the callback is already noop and doesn't get called
 - promise fulfills, callback gets set, callback gets called with the right value for error ✔️ 
```